File: internal/model/song.go
package model
type Song struct {
	ID          string  
	Title       string  `json:"title"`
	Artist      string  `json:"channel"`
	Duration    float64 `json:"duration"`
	Thumbnail   string  `json:"thumbnail"`
	ViewCount   int64   `json:"view_count"`
	UploadDate  string  `json:"upload_date"`
	URL         string  `json:"url"`
	IsLiked bool
	PersonalPlayCount int
	Genre string
}

File: src/models/song.c
#include <stdbool.h>
#include <stdint.h>

typedef int64_t i64 ;

typedef struct {
    char *id;
    char *title;
    char *artist;
    double duration;
    char *thumbnail;
    i64 view_count;
    char *uploaddate;
    char *url;
    bool isliked;
    int personalplaycount;
    char *genre;
} Song;


